src/headless-reducer-manager.ts

```diff
@@ -0,0 +1,52 @@
+import { Reducer, combineReducers } from '@reduxjs/toolkit';
+import { ParentState, State } from './models/state';
+
+import createQuerySlice from './slices/query';
+import createVerticalSlice from './slices/vertical';
+import createUniversalSlice from './slices/universal';
+import createFiltersSlice from './slices/filters';
+import createSpellCheckSlice from './slices/spellcheck';
+import createSessionTrackingSlice from './slices/sessiontracking';
+import createMetaSlice from './slices/meta';
+import createLocationSlice from './slices/location';
+import { ActionWithHeadlessId } from './store';
+
+/**
+ * Manages the current map of headless IDs to Reducers.
+ */
+export default class HeadlessReducerManager {
+  private headlessIdToReducer: Record<string, Reducer> = {};
+
+  addReducer(headlessId: string): void {
+    this.headlessIdToReducer[headlessId] = createAnswersReducer(headlessId + '/');
+  }
+
+  getCombinedReducer(): Reducer<ParentState, ActionWithHeadlessId> {
```

Comment:
nit: could `getParentReducer` be a bit more consistent since we also have a `ParentState`?

Reply:
renamed to getParentReducer and addAnswersReducer!